File: files/300-399/372/CH-372.py
import pandas as pd

path = "300-399/372/CH-372 Filter.xlsx"
input = pd.read_excel(path, usecols="B", skiprows=2, nrows=8)
test = pd.read_excel(path, usecols="F", skiprows=2, nrows=3)

result = input[input['ID'].astype(str).str.contains(r'[13579]')].reset_index(drop=True)

print(result['ID'].equals(test['ID.1']))
# [1] True

File: files/300-399/372/CH-372.R
library(tidyverse)
library(readxl)

path <- "300-399/372/CH-372 Filter.xlsx"
input <- read_excel(path, range = "B3:B10")
test <- read_excel(path, range = "F3:F6")

result = input %>%
  filter(str_detect(ID, "[13579]"))

all.equal(result$ID, test$ID)
# [1] TRUE
